src/analysis/lattice.h

```diff
@@ -0,0 +1,62 @@
+#ifndef wasm_analysis_lattice_h
+#define wasm_analysis_lattice_h
+
+#include "wasm.h"
+#include <bitset>
+#include <iostream>
+
+namespace wasm::analysis {
+
+enum LatticeComparison { NO_RELATION, EQUAL, LESS, GREATER };
+
+template<typename T>
+constexpr bool has_compare = std::is_invocable_r<LatticeComparison,
+                                                 decltype(T::compare),
+                                                 const T&,
+                                                 const T&>::value;
+template<typename T>
+constexpr bool has_getLeastUpperBound = std::
+  is_invocable_r<T, decltype(T::getLeastUpperBound), const T&, const T&>::value;
```

Comment:
for checking member functions, see this working example:

```
#include <type_traits>

struct foo {
  int bar(int, int);
};

static_assert(std::is_invocable_r_v<int, decltype(&foo::bar), foo, int, int>);
```

Note that the method receiver type `foo` is passed to `std::is_invocable` as the first argument, even though that's not really how you call it.

Reply:
Assertions are now run in the monotone analyzer now.